### ISSUE TITLE
ServerSentEventSpec fails with ResourceLeakDetector

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#9731]
### Issue description

Steps to reproduce: 

Micronaut Core branch: 4.0.x
OS: macOS Ventura

```
./gradlew :http-server-netty:test --tests "io.micronaut.http.server.netty.sse.ServerSentEventSpec"
```

It fails more often than passes

https://ge.micronaut.io/scans/tests?search.timeZoneId=Europe/Madrid&tests.container=io.micronaut.http.server.netty.sse.ServerSentEventSpec&tests.test=test%20Content-Type%20header%20is%20correct

```
13:04:16.770 [Test worker] ERROR io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 
#1:
	io.netty.handler.codec.http.DefaultHttpContent.touch(DefaultHttpContent.java:80)
	io.netty.handler.codec.http.DefaultHttpContent.touch(DefaultHttpContent.java:25)
	io.micronaut.http.netty.reactive.HandlerPublisher.publishMessageLater(HandlerPublisher.java:349)
	io.micronaut.http.netty.reactive.HandlerPublisher.channelRead(HandlerPublisher.java:333)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	io.micronaut.http.netty.stream.HttpStreamsHandler.handleReadHttpContent(HttpStreamsHandler.java:276)
	io.micronaut.http.netty.stream.HttpStreamsHandler.channelRead(HttpStreamsHandler.java:254)
	io.micronaut.http.netty.stream.HttpStreamsClientHandler.channelRead(HttpStreamsClientHandler.java:179)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	io.micronaut.http.client.netty.HttpLineBasedFrameDecoder$Wrap.channelRead(HttpLineBasedFrameDecoder.java:102)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	io.netty.handler.codec.ByteToMessageDecoder.handlerRemoved(ByteToMessageDecoder.java:266)
	io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:1122)
	io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637)
	io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:477)
	io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:423)
	io.micronaut.http.client.netty.HttpLineBasedFrameDecoder.channelRead(HttpLineBasedFrameDecoder.java:70)
```

**Comment by sdelamo** [comment]
@yawkat is this something you could check?

**Comment by yawkat** [comment]
i think it's fixed in 4.1

**Comment by sdelamo** [comment]
Confirmed, it does not fail for me in 4.1.
